### ISSUE TITLE
(Blazor.WASM) Cannot use Debug when using generic attribute

### ISSUE THREAD

**Issue by ocSADus** [dotnet/runtime#87871]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When I use some generic attribute in WASM Blazor project, and I tried debug that - breakpoint doesn't work, and I receive a message in console:
"fail: DevToolsProxy[0]
 Failed to load https://localhost:7186/_framework/BlazorApp3.Client.dll (Specified cast is not valid.)"

In the text below, I used a generic attribute in program.cs, but that doesn't matter. You can create and try to use them wherever you want.

Code:
![image](https://github.com/dotnet/aspnetcore/assets/77534200/b7ff26db-461c-4bc6-99a5-40b30859ed70)
Console:
![image](https://github.com/dotnet/aspnetcore/assets/77534200/44092e56-cc80-4458-955b-736f6160ae1f)

### Expected Behavior

Debug is running as usual, and I have worked breakpoints.

### Steps To Reproduce

First of all, you can check repo with this bug: https://github.com/ocSADus/GenericAttributeBug

Steps:

- Open VS 2022
- Click "Create a new project"
- Select "Blazor WebAssembly App"
- At additional Information page select .Net 7.0, ASP.Net Core Hosted, Do not use top-level statements
- Open Program.cs for client side
- Create Generic Attribute and use them (look at my screenshot for example)
- Set breakpoint
- Try to debug app

### Exceptions (if any)

fail: DevToolsProxy[0]
 Failed to load https://localhost:7186/_framework/BlazorApp3.Client.dll (Specified cast is not valid.)

### .NET Version

7.0.302

### Anything else?

Target framework: .NET 7.0
Language version: 11.0
IDE: VS 2022 17.6.2

--info:
.NET SDK:
 Version:   7.0.302
 Commit:    990cf98a27

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.302\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

**Comment by ilonatommy** [comment]
Yes, a duplicate of https://github.com/dotnet/runtime/issues/86765. Backporting to net 7 was done here: https://github.com/dotnet/runtime/pull/86862.

**Comment by benm-eras** [comment]
I have just found myself blocked by this. @ilonatommy you mentioned that the fix has been backported to .NET 7; how and when would I get this fix?

**Comment by ilonatommy** [comment]
@lewing, could you help here? 7.0.8 has info "Latest release date June 22, 2023" so that's too early, the change is for sure not there. Should it be 7.0.9?

**Comment by benm-eras** [comment]
just FYI I have the 7.0.305 version of the SDK which includes the 7.0.8 runtime and still have the issue.

**Comment by thaystg** [comment]
Is this one right? https://github.com/dotnet/runtime/pull/86862 this is on 7.0.8.
@benm-eras did you upgrade the nuget packages to use 7.0.8?

**Comment by benm-eras** [comment]
@thaystg I didn't think to do that! I updated updated all my nuget packages. Although there were others, I imagine the significant ones were the `Microsoft.AspNetCore.Components.WebAssembly.*` packages which were on 7.0.8 and are now on 7.0.9. Any way, client side debugging is now working as expected. 

To summarise, I updated my SDK to 7.0.305 (and therefore runtime to 7.0.8) and the `Microsoft.AspNetCore.Components.WebAssembly.*` Nuget packages to 7.0.9.

Thanks you for your support!
